subsys/bluetooth/shell/bt.c

```diff
@@ -1122,6 +1122,40 @@ static int cmd_bondable(const struct shell *shell, size_t argc, char *argv[])
 	return 0;
 }
 
+static void bond_info(const struct bt_bond_info *info,void *user_data)
+{
+	int *bond_count = (int *)user_data;
+	char addr[BT_ADDR_LE_STR_LEN];
+	bool connected = false;
+	struct bt_conn * conn;
+
+	bt_addr_le_to_str(&info->addr, addr, sizeof(addr));
+
+	conn = bt_conn_lookup_addr_le(selected_id, &info->addr);
+	if (conn) {
+		connected = true;
+		bt_conn_unref(conn);
+	}
+
+	shell_print(ctx_shell, "Remote Identity: %s, status: %s", addr,
+		    connected ? "Connected" : "Disconnected");
```

Comment:
This isn't completely correct, since `bt_conn_lookup_addr_le()` may return unconnected objects as well, in case some piece of code is hanging onto their `bt_conn` reference even though the state of the connection has become disconnected.

Reply:
True. Switched to only printing the conn object pointer instead. Gives an indication to the state, although not sure if it is needed.

Reply:
It seems a bit arbitrary to me. Don't we have a separate command to list connections? (if we don't perhaps we should add one?)

Reply:
We don't have one for connections. I can add one.
I wanted to give the user an indications of which bonds where connected or not. So having a connections command to print similar info as this command would achieve that.